### PR TITLE
Partner Experience: horizon containerized runtime suggestions

### LINKED ISSUE #142: Horizon containerization best practices
### What problem does your feature solve?
Partner Experience initiative - Not much direction is provided in [Running docs](https://developers.stellar.org/docs/run-api-server/running) for how best to configure horizon and it's supporting infrastructure as containers, rather settings are documented individually and left to user to figure out best combinations.

### What would you like to see?
Provide updated copy under the [Running section](https://developers.stellar.org/docs/run-api-server/running), it should provide opinionated directions(and brief reason why) to informs reader on how they can deploy horizon in containers:

What should the containerization boundaries look like, should postgresql be internal to the same container as horizon?
Should a watcher core node be in same container as horizon? 
How should command line parameters be passed to the horizon container, as environment variables?
When, where and how should shared external volumes be defined on any containers to enable larger storage requirements(most containers only get like 32G or less for default ephemeral storage).
 
Update the dockerfiles at https://github.com/stellar/go/blob/master/services/horizon/docker, consider adding a new `examples` directory with new Dockerfile(s) that codify the recommendations here in docs.

### PR DESCRIPTION
**Pull request by sreuland**
Closes #142 

Reference the new installing and configuring content as a preface before describing the how-to run content and go into detail on the containerized runtime options further. make sure install/configure/run are all consistent with each other but not overlapping on content.

best way to see the changes is the temporary published docs page for running horizon - https://developers-pr217.previews.kube001.services.stellar-ops.com/docs/run-platform-server/running

### CODE REVIEW

File: docs/run-platform-server/configuring.mdx [hunk 1]
```diff
@@ -8,7 +8,14 @@ import { CodeExample } from "@site/src/components/CodeExample";
 
 Pre-requisites:
 
-- [Installation](./installing.mdx) is completed on a host machine and results in two executables: `stellar-horizon` and `stellar-core`. Per the installation instructions, `stellar-core` should be included in your host OS' `$PATH`, as the configurations below result in Horizon searching for a `stellar-core` to execute.
+- You have identified the [Installation](./installing.mdx) method for host system.
+
+  For bare metal, you have the two executables: `stellar-horizon` and `stellar-core` both on host OS' `$PATH`.
+
+  For running image, using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`, you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon#<tag>` onto host.
```
Comment (urvisavla) [inline docs/run-platform-server/configuring.mdx:15]:
```suggestion
  For running image, using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`, you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon#<tag>` onto host.
```
```suggestion
  For running image, using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`, you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon:<tag>` onto host.
```

Comment (urvisavla) [reply docs/run-platform-server/configuring.mdx:15]: >  using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`

Not sure I understood this. Are we saying that the docker run will also pull the image?

> you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon#<tag>` onto host.

Would it be helpful to include the docker pull command in the installing doc?

Comment (urvisavla) [inline docs/run-platform-server/configuring.mdx:15]:
```suggestion
  For running image, using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`, you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon#<tag>` onto host.
```
```suggestion
  For running image, using the [stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) direct on `docker run`, you have pulled the stellar/stellar-horizon image via `docker pull stellar/stellar-horizon:<tag>` onto host.
```

**Comment by sreuland** [comment]
> Would it be helpful to include the docker pull command in the installing doc?

yes, I adjusted that to consistently show the docker pull aspect - https://github.com/stellar/stellar-docs/pull/217/commits/6ead9756fac44c8e4dffa1174a1a93688c5fc11a, thanks!

File: docs/run-platform-server/installing.mdx [hunk 2]
```diff
@@ -9,10 +9,10 @@ To install Horizon in production or non-development environments, we recommend t
...
-  - Helm charts, when the target envrionment uses container orchestration such as Kubernetes and has enabled Helm Charts on cluster. The [Horizon Helm chart](https://github.com/stellar/helm-charts/tree/main/charts/horizon) manages installation life cycle. Use the [Helm Install command](https://helm.sh/docs/helm/helm_install/), it will accept Horizon's configuration parameters. Please review [Configuration](./configuring.mdx) first, to identify any specific configuration params needed.
+  - non-Helm Chart, if the target environment for container does not support Helm chart installation, pull the prebuilt image [dockerhub.com/stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) with docker daemon on host. Choose the tag of that image for specific release version and then pull with `docker pull stellar/stellar-horizon#<tag>` onto host.
+  - Helm charts, when the target envrionment uses container orchestration such as Kubernetes. The [Horizon Helm chart](https://github.com/stellar/helm-charts/tree/main/charts/horizon) manages installation life cycle.
```
Comment (aditya1702) [inline docs/run-platform-server/installing.mdx:15]:
```suggestion
  - **Helm Charts**: when the target envrionment uses container orchestration such as Kubernetes. The [Horizon Helm chart](https://github.com/stellar/helm-charts/tree/main/charts/horizon) manages installation life cycle.
```

**Review by aditya1702 — CHANGES_REQUESTED** [review]

File: docs/run-platform-server/configuring.mdx (hunk 1 above)
Comment (sreuland) [reply docs/run-platform-server/configuring.mdx]: fixed, thx! - https://github.com/stellar/stellar-docs/pull/217/commits/ee97a781751662040d7b5897078f78ce89c1134d

File: docs/run-platform-server/installing.mdx [hunk 3]
```diff
@@ -9,10 +9,10 @@ To install Horizon in production or non-development environments, we recommend t
...
 - containerized:
-  - non-Helm Chart, if the target envrironment for container to run does not support Helm chart usage, run the prebuilt docker image of Horizon published on [dockerhub.com/stellar/horizon](https://hub.docker.com/r/stellar/stellar-horizon).
-  - Helm charts, when the target envrionment uses container orchestration such as Kubernetes and has enabled Helm Charts on cluster. The [Horizon Helm chart](https://github.com/stellar/helm-charts/tree/main/charts/horizon) manages installation life cycle. Use the [Helm Install command](https://helm.sh/docs/helm/helm_install/), it will accept Horizon's configuration parameters. Please review [Configuration](./configuring.mdx) first, to identify any specific configuration params needed.
+  - non-Helm Chart, if the target environment for container does not support Helm chart installation, pull the prebuilt image [dockerhub.com/stellar/stellar-horizon](https://hub.docker.com/r/stellar/stellar-horizon) with docker daemon on host. Choose the tag of that image for specific release version and then pull with `docker pull stellar/stellar-horizon#<tag>` onto host.
```
[the remark this answers is not in the document]
Comment (sreuland) [reply docs/run-platform-server/installing.mdx]: I re-worked this, used Orchestrated vs. Non-Orchestrated terminology instead.

File: docs/run-platform-server/installing.mdx (hunk 2 above)
Comment (sreuland) [reply docs/run-platform-server/installing.mdx]: I  re-worked this with Orchestrated and Non-Orchestrated usage instead, let me know if that reads better.

**Comment by sreuland** [comment]
@aditya1702 , @urvisavla , I've responded to your comments, can you re-check, thanks!

**Pull request merged**